library/manageiq_provider.py

```diff
@@ -164,6 +169,48 @@ def __init__(self, module, url, user, password, verify_ssl, ca_bundle_path):
         self.changed       = False
         self.providers_url = self.api_url + '/providers'
 
+    def auths_validation_details(self, provider_id):
+        try:
+            result = self.client.get('{providers_url}/{id}/?attributes=authentications'.format(providers_url=self.providers_url, id=provider_id))
+            auths = result.get('authentications', [])
+            return {auth['authtype']: auth for auth in auths}
+        except Exception as e:
+            self.module.fail_json(msg="Failed to get provider data. Error: %s" % e)
+
+    def verify_authenticaion_validation(self, provider_id, old_validation_details, authtypes_to_verify):
+        """ Verifies that the provider's authentication validation passed.
+        provider_id            - the provider's id manageiq
+        old_validation_details - a tuple of (last_valid_on, last_invalid_on), representing the last time
+                                 that the authentication validation occured (success or failure).
+        authtypes_to_verify    - a list of autentication types that require validation
+
+        Returns a (success, details) tuple:
+            success: True if authentication validation passed, False otherwise
+            details: 'All Valid' if passed, authentication validation details otherwise
+        """
+        for i in range(ManageIQ.ITERATIONS):
+            new_validation_details = self.auths_validation_details(provider_id)
+
+            def validated(old, new):
+                """ Returns True if the validation timestamp, valid or invalid, is different
+                from the old validation timestamp, False otherwise
+                """
+                return ((old.get('last_valid_on'), old.get('last_invalid_on')) !=
+                        (new.get('last_valid_on'), new.get('last_invalid_on')))
+
+            validations_done = all(validated(old_validation_details.get(t, {}), new_validation_details.get(t, {}))
+                                   for t in authtypes_to_verify)
+            details = {t: (new_validation_details[t].get('status', "Validation in progress"),
+                           new_validation_details[t].get('status_details', ''))
+                       for t in authtypes_to_verify}
+            if validations_done:
+                if any(new_validation_details[t]['status'] not in ('Valid', None) for t in authtypes_to_verify):
+                    return False, details
+                if all(new_validation_details[t]['status'] == 'Valid' for t in authtypes_to_verify):
+                    return True, 'All Valid'
+            time.sleep(ManageIQ.WAIT_TIME)
+        return False, details
```

Comment:
I think the details returned here might include misleading old `status`, `status_details`.
Should we omit status whose timestamps indicate they're old?  OTOH we're here *because* validation didn't complete yet (or never started for some reason), and we want *some* info on ones who didn't complete.